docs for /api/sendMail (#390)
sofn-xyz/mailing: packages/web/pages/docs/rest-api.mdx

## Send and Render Emails

> | Method    | Path                           | Description                          |                                                                                                      Example |

Reviewer: ```suggestion
| Method    | Path                           | Description                          |                                                                                                      Details |
```

- Include the API key in the request headers: `X-API-KEY=...`
- Include the API key as a query string parameter: `?apiKey=...`

> ## API endpoints

Reviewer: ```suggestion
## API Details
```

- Include the API key in the request headers: `X-API-KEY=...`
- Include the API key as a query string parameter: `?apiKey=...`

## API endpoints

All API endpoints expect the body to be a JSON string and the header Content-Type to be "application/json".

### /api/sendMail

Send an email using your configured sendmail.

**Authentication required**

Yes, using an [API key](#api-keys)

**Request parameters**

You can either use /api/sendMail to send raw html or a template.

> - [Standard nodemailer options](#standard-nodemailer-options)

Reviewer: I think it would be better to reference https://www.mailing.run/docs/sending-email#calling-sendmail  than nodemailer docs. The nodemailer section at the end feels very random unless you get there from this link.

- templateName: string (required if using template mode, must be the name of one of your templates)
- props: []string (required if using template mode, must correspond to the props your template expects)
- html: string (required if using raw html mode)

> - previewName: string (optional) - used for analytics only, gets put on the Message object in the database so you can tell what preview was used to send the email

Reviewer: ```suggestion
- previewName: string (optional) - this gets put on the Message object in the database so you have a category beyond templateName for the type of email being sent. Useful for segmenting analytics if the templateName is more general.
```

Example request (template mode):

```
POST /api/sendMail
Content-Type: application/json
X-API-KEY: your-api-key

{
  "to": "to@example.com",
  "from": "you@example.com",
  "subject": "Hello world",
  "templateName": "myTemplate",
  "props": {
    "name": "Alex"
  },
}

> ```

Author: Yeah, that would be cool

> ```

Author: @psugihara I played with this a bit in another branch called `382-doc-api-sendmail-code` I think that's how it could work:
<img width="1757" alt="Screenshot 2022-12-01 at 8 37 31 PM" src="https://user-images.githubusercontent.com/24866/205215960-5652c475-be00-47e6-8b7d-d427e77e7636.png">

Do you know how to get the rehype/highlight.js syntax highlighting working?  I see that it works if you use ```js in the markdown, but prettier then squashes the formatting of the code, so I need to apply it manually.  But we could easily do that in CodeBlock (in that branch) where we render the text.  I think I almost got it working, but confused about what is necessary for it -- I see that we're using rehype-highlight, but what do we need to use that outside of the context of markdown?  Probably some sub-dependency of it.

> ```

Author: I figured out a way to do it without doing prettier ignore and keeping the code the same way otherwise (in the mdx etc)

Example request (raw html mode):

```
POST /api/sendMail
Content-Type: application/json
X-API-KEY: your-api-key

{
  "to": "to@example.com",
  "from": "you@example.com",
  "subject": "Hello world",
  "html": "<html><body>Hello World</body></html>",
  "text": "Hello world",

> }

Author: agree

```

**Success response**

HTTP Status Code 200
Response body: JSON object with possible attributes

> - result: [SMTPPool.SentMessageInfo](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6f93d358c4eb1d30bf0da65606c27c58a1a0979a/types/nodemailer/lib/smtp-pool/index.d.ts#L35) the return value from sendMail

Reviewer: Example of a success response would be good here.

> - result: [SMTPPool.SentMessageInfo](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6f93d358c4eb1d30bf0da65606c27c58a1a0979a/types/nodemailer/lib/smtp-pool/index.d.ts#L35) the return value from sendMail

Author: ✅

**Error responses**
HTTP Status Code: 401 or 422 or 500

401 indicates an invalid API key

422 indicates an invalid request

Response body: JSON object with possible attributes

- error: string - error message
- mjmlErrors: []string - MJML errors thrown during template rendering

500 indicates an internal server error

- error: string - error message

Example error response:

```
422 Unprocessable Entity
Content-Type: application/json

{
  "error": "sendMail couldn't find a subject for your email"
}
```

> ## Standard nodemailer options

Reviewer: ref https://www.mailing.run/docs/sending-email#calling-sendmail and remove
